h2/src/main/org/h2/command/Parser.java

```diff
@@ -4859,13 +4866,20 @@ private String readIdentifierWithSchema(String defaultSchemaName) {
 
     private String readIdentifierWithSchema2(String s) {
         schemaName = s;
-        s = readColumnIdentifier();
-        if (currentTokenType == DOT) {
-            if (equalsToken(schemaName, database.getShortName())) {
-                read();
-                schemaName = s;
+        if (readIf(DOT)) {
```

Comment:
Why this method is so complicated now?

Reply:
I tried to support catalog..tablename where the schemaname gets left out

Reply:
SQL Standard does not allow such chains of identifiers with missing parts in the middle. H2 should reject them too.

Reply:
I searched how often this is used in our code, and it is. Since it is officially allowed in sybase and I think in mssql as well, I would like to keep it supported. I also tried this in postgres, there it is not possible.

Reply:
thank for doing the research. We prefer not to allow non-standard stuff like this, unless it is hidden behind a mode flag. Please have a look at the mode class, add a flag there, set that flag to true for MSSQL, and check that flag here

Reply:
see org.h2.test.db.TestIgnoreCatalogs.doesNotAcceptEmptySchemaWhenNotMSSQL()

Reply:
This method is partially broken with default settings.

```SQL
SELECT * FROM PUBLIC..TEST;
```
throws _Table "PUBLIC" not found_. Before your changes the error message was better.

Reply:
Is a new ErrorCode necessary, or what would you prefer?

Reply:
Just restore the previous behavior when the new setting / compatibility mode aren't activated.

Reply:
Should be fixed by the most recent commit.